src/ui/controllercollection.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Manages UI Controllers.
+ *
+ * @class ControllerCollection
+ * @extends Collection
+ */
+CKEDITOR.define( [
+	'collection',
+	'ckeditorerror'
+], ( Collection, CKEditorError ) => {
+	class ControllerCollection extends Collection {
```

Comment:
Or without passing `this` to `ControllerCollection` – the controller could listen on the `ControllerCollection#add/remove` events.